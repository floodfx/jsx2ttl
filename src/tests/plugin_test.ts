import { expect, test } from "bun:test";

// test snapshots of the generated code
// these tests rely on the jsx2ttlPlugin being loaded via bunfig.toml

import Profile from "../jsx/simple_static";
test("simple_static", () => {
  expect(Profile().toString()).toMatchSnapshot();
});

import Gallery from "../jsx/nested_static";
test("nested_static", () => {
  expect(Gallery().toString()).toMatchSnapshot();
});

import CWithProps from "../jsx/simple_props";
test("simple_props", () => {
  expect(CWithProps({count: 11}).toString()).toMatchSnapshot();
});

import App from "../jsx/nested_props";
test("nested_props", () => {
  expect(App({name: "bar"}).toString()).toMatchSnapshot();
});


