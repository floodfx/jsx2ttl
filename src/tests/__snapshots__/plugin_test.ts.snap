// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`simple_static 1`] = `"<img src="https://i.imgur.com/MK3eW3Am.jpg" alt="Katherine Johnson">"`;

exports[`simple_props 1`] = `"<div>hi 11!</div>"`;

exports[`nested_static 1`] = `
"<section>
      <h1>Amazing scientists</h1>
      <img src="https://i.imgur.com/MK3eW3Am.jpg" alt="Katherine Johnson">
      <img src="https://i.imgur.com/MK3eW3Am.jpg" alt="Katherine Johnson">
      <img src="https://i.imgur.com/MK3eW3Am.jpg" alt="Katherine Johnson">
    </section>"
`;

exports[`nested_props 1`] = `"<h1 className="foo">Hello bar. <div>hi 4</div></h1>"`;
